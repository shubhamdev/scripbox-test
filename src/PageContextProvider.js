import React, { useState, createContext } from "react";
export const PageContext = createContext();
const PageContextProvider = (props) => {
  const data = useState("");
  return (
    <PageContext.Provider
      value={{
        user: data,
      }}
    >
      {props.children}
    </PageContext.Provider>
  );
};
export default PageContextProvider;
