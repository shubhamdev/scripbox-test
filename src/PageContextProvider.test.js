import React from "react";
import { unmountComponentAtNode } from "react-dom";
import { render, screen } from "@testing-library/react";
import { shallow } from "enzyme";
import PageContextProvider from "./PageContextProvider";

let container = null;

beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

describe("PageContextProvider", () => {
  it('should render PageContextProvider correctly in "debug" mode', () => {
    const component = shallow(<PageContextProvider debug />);
    expect(component).toMatchSnapshot();
  });
  it("should render PageContextProvider correctly with no props", () => {
    const component = shallow(<PageContextProvider />);
    expect(component).toMatchSnapshot();
  });
});
