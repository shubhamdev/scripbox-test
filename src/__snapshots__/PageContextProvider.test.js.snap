// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageContextProvider should render PageContextProvider correctly in "debug" mode 1`] = `
<ContextProvider
  value={
    Object {
      "user": Array [
        "",
        [Function],
      ],
    }
  }
/>
`;

exports[`PageContextProvider should render PageContextProvider correctly with no props 1`] = `
<ContextProvider
  value={
    Object {
      "user": Array [
        "",
        [Function],
      ],
    }
  }
/>
`;
